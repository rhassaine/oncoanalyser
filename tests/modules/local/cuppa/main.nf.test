nextflow_process {
    name "Test Process CUPPA"
    script "modules/local/cuppa/main.nf"
    process "CUPPA"
    config "tests/nextflow.config"

    test("CUPPA with default parameters (ALL categories)") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor', sample_rna_id: 'subject_a.tumor_rna'],
                    file('/home/rhassaine/container_test_output/subject_a/isofox/'),
                    file('/home/rhassaine/container_test_output/subject_a/purple/'),
                    file('/home/rhassaine/container_test_output/subject_a/linx/somatic_annotations/'),
                    file('/home/rhassaine/container_test_output/subject_a/virusinterpreter/')
                ]
                input[1] = 38
                input[2] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/misc/cuppa/alt_sj.selected_loci.38.tsv.gz", checkIfExists: true)
                input[3] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/misc/cuppa/cuppa_classifier.38.pickle.gz", checkIfExists: true)
                input[4] = 'ALL'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.cuppa_dir != null },
                { assert process.out.cuppa_dir.size() > 0 },
                { assert snapshot(process.out).match() }
            )

            if (process.success) {
                def cuppa_output_dir = process.out.cuppa_dir[0][1]
                def sample_id = process.out.cuppa_dir[0][0].sample_id
                assertAll(
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa_data.tsv.gz").exists() },
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa.pred_summ.tsv").exists() },
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa.vis_data.tsv").exists() },
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa.vis.png").exists() },
                    { assert process.out.versions != null },
                    { assert process.out.versions.size() > 0 },
                    { assert path(process.out.versions[0]).exists() }
                )
            }
        }
    }

    test("CUPPA with DNA-only categories") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor', sample_rna_id: 'subject_a.tumor_rna'],
                    [],  // No ISOFOX directory for DNA-only
                    file('/home/rhassaine/container_test_output/subject_a/purple/'),
                    file('/home/rhassaine/container_test_output/subject_a/linx/somatic_annotations/'),
                    file('/home/rhassaine/container_test_output/subject_a/virusinterpreter/')
                ]
                input[1] = 38
                input[2] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/misc/cuppa/alt_sj.selected_loci.38.tsv.gz", checkIfExists: true)
                input[3] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/misc/cuppa/cuppa_classifier.38.pickle.gz", checkIfExists: true)
                input[4] = 'DNA'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.cuppa_dir != null },
                { assert process.out.cuppa_dir.size() > 0 },
                { assert snapshot(process.out).match() }
            )

            if (process.success) {
                def cuppa_output_dir = process.out.cuppa_dir[0][1]
                def sample_id = process.out.cuppa_dir[0][0].sample_id
                assertAll(
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa_data.tsv.gz").exists() },
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa.pred_summ.tsv").exists() },
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa.vis_data.tsv").exists() },
                    { assert path(cuppa_output_dir).resolve("${sample_id}.cuppa.vis.png").exists() },
                    { assert process.out.versions != null },
                    { assert process.out.versions.size() > 0 },
                    { assert path(process.out.versions[0]).exists() }
                )
            }
        }
    }

    test("CUPPA should fail when input is missing") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor', sample_rna_id: 'subject_a.tumor_rna'],
                    [],  // Missing ISOFOX directory
                    [],  // Missing PURPLE directory
                    [],  // Missing LINX directory
                    []   // Missing VIRUSINTERPRETER directory
                ]
                """
            }
        }

        then {
            assertAll(
                { assert !process.success },
                { assert process.exitStatus != 0 }
            )
        }
    }
}