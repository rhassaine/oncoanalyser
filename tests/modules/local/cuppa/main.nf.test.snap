{
    "CUPPA with default parameters (ALL categories)": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor",
                            "sample_rna_id": "subject_a.tumor_rna"
                        },
                        [
                            "subject_a.tumor.cuppa.pred_summ.tsv:md5,fc33c20e3766cdd7ee64a8062e864686",
                            "subject_a.tumor.cuppa.vis.png:md5,cf4703e244012ca96f37632a54bdcc7a",
                            "subject_a.tumor.cuppa.vis_data.tsv:md5,2237901c760e5a7ca3d3bbd1f0781132",
                            "subject_a.tumor.cuppa_data.tsv.gz:md5,f0291c0e9969db16a6ba9d1cab5e8860"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,4b4bb425530b387ae7d8821332a5f609"
                ],
                "cuppa_dir": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor",
                            "sample_rna_id": "subject_a.tumor_rna"
                        },
                        [
                            "subject_a.tumor.cuppa.pred_summ.tsv:md5,fc33c20e3766cdd7ee64a8062e864686",
                            "subject_a.tumor.cuppa.vis.png:md5,cf4703e244012ca96f37632a54bdcc7a",
                            "subject_a.tumor.cuppa.vis_data.tsv:md5,2237901c760e5a7ca3d3bbd1f0781132",
                            "subject_a.tumor.cuppa_data.tsv.gz:md5,f0291c0e9969db16a6ba9d1cab5e8860"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4b4bb425530b387ae7d8821332a5f609"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T04:10:17.027766304"
    },
    "CUPPA with DNA-only categories": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor",
                            "sample_rna_id": "subject_a.tumor_rna"
                        },
                        [
                            "subject_a.tumor.cuppa.pred_summ.tsv:md5,ddca4a0507895bee001f855ea8e0d061",
                            "subject_a.tumor.cuppa.vis.png:md5,99d4dcf7263b63cbd048c8cdba1b085a",
                            "subject_a.tumor.cuppa.vis_data.tsv:md5,9c00514c21ad934b5ad328d01515b547",
                            "subject_a.tumor.cuppa_data.tsv.gz:md5,b2894129214198df6bde85d532f0f9b8"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,4b4bb425530b387ae7d8821332a5f609"
                ],
                "cuppa_dir": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor",
                            "sample_rna_id": "subject_a.tumor_rna"
                        },
                        [
                            "subject_a.tumor.cuppa.pred_summ.tsv:md5,ddca4a0507895bee001f855ea8e0d061",
                            "subject_a.tumor.cuppa.vis.png:md5,99d4dcf7263b63cbd048c8cdba1b085a",
                            "subject_a.tumor.cuppa.vis_data.tsv:md5,9c00514c21ad934b5ad328d01515b547",
                            "subject_a.tumor.cuppa_data.tsv.gz:md5,b2894129214198df6bde85d532f0f9b8"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4b4bb425530b387ae7d8821332a5f609"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T04:10:34.877952937"
    }
}